internal/transformer/classes/analysers/springbootanalyser.go

```diff
@@ -88,104 +107,400 @@ func (t *SpringbootAnalyser) BaseDirectoryDetect(dir string) (namedServices map[
+				isTomcatProvided = true
+			}
 		}
 	}
-	if !isSpringboot {
-		logrus.Debugf("Ignoring pom at %s as does not contain Springboot dependencies: org.springframework.boot", dir)
-		return nil, nil, nil
-	}
 
-	// Collect packaging
 	packaging := ""
 	if pom.Packaging == "" {
-		logrus.Debugf("Pom at %s does not contain a Packaging block", dir)
+		logrus.Debugf("Pom at %s does not contain a Packaging block", pomXMLPath)
 	} else {
 		packaging = pom.Packaging
 		logrus.Debugf("Packaging: %s", packaging)
 	}
 
 	// Collect java / tomcat version fom the Properties block
 	javaVersion := ""
-	tomcatVersion := ""
+	//tomcatVersion := ""
 	if pom.Properties == nil {
-		logrus.Debugf("Pom at %s  does not contain a Properties block", dir)
+		logrus.Debugf("Pom at %s  does not contain a Properties block", pomXMLPath)
 	} else {
+
 		for k, v := range pom.Properties.Entries {
 			switch k {
+			// Only for springboot apps
 			case "java.version":
 				javaVersion = v
-			case "tomcat.version":
-				tomcatVersion = v
+			//case "tomcat.version":
+			//	tomcatVersion = v
+			// Non springboot apps:
+			case "maven.compiler.target":
+				javaVersion = v
 			}
 		}
 	}
-	logrus.Debugf("Java version %s", javaVersion)
-	logrus.Debugf("Tomcat version %s", tomcatVersion)
 
-	// Check if the application uses an embeded server or not. If not, identify which server
-	isServerEmbedded := false
+	conf := Configuration{
+		BuildTool:        "maven",
+		HasModules:       hasModules,
+		IsSpringboot:     isSpringboot,
+		IsTomcatProvided: isTomcatProvided,
+		Packaging:        packaging,
+		JavaVersion:      javaVersion,
+		Name:             pom.Name,
+		ArtifactID:       pom.ArtifactID,
+		Version:          pom.Version,
+	}
 
-	// First condition: Tomcat with provided scope?
-	isTomcatProvided := false
-	for _, dependency := range *pom.Dependencies {
-		if strings.Contains(dependency.ArtifactID, "spring-boot-starter-tomcat") && dependency.Scope == "provided" {
-			isTomcatProvided = true
+	return conf, nil
+
+}
+
+// DirectoryDetect runs detect in each sub directory
+func (t *SpringbootAnalyser) DirectoryDetect(dir string) (namedServices map[string]transformertypes.ServicePlan, unnamedServices []transformertypes.TransformerPlan, err error) {
+	destEntries, err := ioutil.ReadDir(dir)
+	if err != nil {
+		logrus.Errorf("Unable to process directory %s : %s", dir, err)
+		return nil, nil, err
+	}
+	mavenFound := false
+	gradleFound := false
+	for _, de := range destEntries {
+		if de.Name() == pomXML {
+			mavenFound = true
+			continue //break
+		}
+		if de.Name() == buildGradle {
+			gradleFound = true
+			continue
 		}
 	}
-	// Second condition: Is packaging WAR
+
+	// If there are not build config files, we stop
+	if !mavenFound && !gradleFound {
+		return nil, nil, nil
+	}
+
+	//config := Configuration{}
+	var config Configuration
+	if mavenFound {
+		mavenConfig, err := GetMavenData(filepath.Join(dir, pomXML))
```

Comment:
do we want to do all this processing in detect or is it better to move them to transform, so that we can interact with the user for doubts. In detect, we can just detect 2 services one for maven and another for gradle, and based on whichever is chosen, the transform step can do the rest of parsing.